File: android/src/main/kotlin/com/hodoan/ble_sdk/event/ScanResultEvent.kt
package com.hodoan.ble_sdk.event

import android.os.Looper
import io.flutter.plugin.common.EventChannel

class ScanResultEvent : EventChannel.StreamHandler {
    private var sink: EventChannel.EventSink? = null

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        sink = events
    }

    override fun onCancel(arguments: Any?) {
        sink = null
    }

    fun success(value: com.hodoan.ble_sdk.ProtobufModel.BluetoothBLEModel) {
        android.os.Handler(Looper.getMainLooper()).post { sink?.success(value.toByteArray()) }
    }
}

File: ios/Classes/Event/ScanResultEvent.swift
//
//  ScanResultEvent.swift
//  ble_sdk
//
//  Created by Doan Ho on 10/05/2023.
//

import Foundation
import Flutter

class ScanResultEvent: NSObject, FlutterStreamHandler{
    private var sink:FlutterEventSink? = nil
    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        sink = events
        return nil
    }
    
    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        sink = nil
        return nil
    }
    
    func sendData(_ data: BluetoothBLEModel) {
        sink?(try? data.serializedData())
        return
    }
}
